### DISCUSSION TITLE
modeling a thermocamera

### DISCUSSION THREAD

**Discussion by fr-evtech** [firemodels/fds#13684] [Q&A] [answered]
Good morning community,
I read a previous discussion from June about the same topic, but I didn't understand how to implement the solution, any help you can provide is more than welcome!
I need to model a few thermocameras, each one of them will be installed on top of a dump, on the separation walls; the purpose is an early detection, which will activate a control related to the smoke evacuation system and so on. If I correctly understood the mechanism, the cameras are supposed to record the radiation PUA (kW/m2) and then an algorithm converts the result to a temperature value.

I need help with the following procedures:
1- in the June discussion it's suggested to use a large number of radiometers, which are solid phase devices, but where should I put them geometry-wise and how many of them?
2- to establish the radiometer's setpoint in kW/m2, should I fix the desired alarm temperature of the dump surface (which I don't know at the moment) and then calculate the radiation with the Stefan Boltzmann law?

Below an image of the current geometry
![Screenshot 2024-11-04 142346](https://github.com/user-attachments/assets/b5000c4b-aa73-44bd-82d4-9372cd853875)

Thank you everybody and sorry for the grammar mistakes, english is not my first language.

**Comment by Er9y714** [comment]
Thermal cameras give temperature output. This depends on the hot surface/flame temperature, emitting body emissivity and transmissivity of the air. If you use radiometer, it will receive the measurement from a large angle, therefore the output might be low (low view factor). Whereas, the sensors in the thermal camera has small angle for higher resolution.

**Reply by fr-evtech** [reply]
So radiometer is not the right device? What should I use?

**Comment by drjfloyd** [comment]
If you are looking to have FDS predict the image a camera sees, I don't think this will be possible. FDS is only going to resolve the radiation field at the level of a grid cell. Your camera sensor is going to be mm across vs. likely mulitple cm for the grid size.

**Reply by fr-evtech** [reply]
That's not what I need cameras to do, what I need is them to activate whenever the radiation (=temperature through stefan-boltzmann law?) reach a setpoint, so they trigger a control in FDS to open/close doors, etc.
Unfortunately I cannot place another, more common type of sensor (smoke/heat detector, beam detector) since there will only be that type of thermocamera for fire-safety purpose...

**Reply by Er9y714** [reply]
@fr-evtech If the thermal camera resolution is high enough, would not it detect a candle flame from 10 m away? Then would you require your FDS model to detect it at 100 W candle flame size?

**Reply by drjfloyd** [reply]
In FDS, your angular resolution is set by NUMBER_RADIATION_ANGLES. This is 100 by default which winds up being 104 actual angles to get relatiely equal solid angles of over the unit sphere. Depending on how many pixels your camera has and how many it would require to activate an alarm, the equivalent FDS angles could be in the 100s of thousands or more. 

To approximate a camera you could place a large number of RADIATIVE HEAT FLUX GAS devices and given them all a view angle representing the angular width of your solid angle. You will probably need to crank up the NUMBER_OF_RADIATION_ANGLES if you want any given gauge to see only a small portion of the domain.

**Reply by fr-evtech** [reply]
@drjfloyd ok I have to admit I don't know anything about NUMBER_RADIATION_ANGLES but it sounds really interesting, I looked for it on the User Guide but I didn't find a lot...is there another good source where I could learn from?

About gas devices, which could be an optimal solution, how do I set the view angle? Is it a variable linked directly to NUMBER_RADIATION_ANGLES?

**Reply by drjfloyd** [reply]
The Technical Reference Guide discusses in detail the radiation transport model in FDS. For the view angle, try searching the user guide for view angle.

**Reply by fr-evtech** [reply]
I literally tried to search "view angle" before asking, but got zero correspondence :( single words have too many instead...
I'll go through it, thanks for your help!

**Comment by Vojta-Salek** [comment, marked as answer]
Wouldn't it be sufficient to use surface temperature as an FDS output?

As mentioned above, a thermocamera provides surface temperatures based on simple calculations using predefined properties (e.g., emissivity, air properties). I think, there is in general considerable uncertainty in the thermocamera outputs themselves. Perhaps allowing for a sufficiently high margin in the limit temperature (considered to trigger the alarm) would be adequate..? Yes, you will ultimately encounter the same issue with mesh resolution, but at least this approach avoids the need to solve for high-resolution radiation.

Am I completely wrong about this? (It's highly possible!)

**Reply by fr-evtech** [reply]
Yes, I actually considered putting a solid phase device on the top of the burner to measure the adiabatic surface temperature, but I was unsure... I didn't follow you when you mentioned mesh resolution though, why will that be a problem?

**Reply by Vojta-Salek** [reply]
Well, if you represent a candle flame using 10 cm thick computational cells, the very high (almost point source) temperature will be averaged across the entire 10 x 10 x 10 cm volume. This averaging effect smooths out high-intensity, localized heat sources, effectively reducing the peak temperatures observed in the simulation output. A similar effect occurs for surface temperatures, as they are averaged across the whole 10 x 10 cm surface.

Unless the entire surface is exposed to uniform heat, the averaging effect at the surface will effectively lower the temperatures you observe in the simulation. This is particularly noticeable in scenarios like radially spreading fires, where different parts of the surface receive varying heat intensities.

**Reply by fr-evtech** [reply]
Ok I understand now, I was actually thinking of some other problem related to mesh dimension and trasmission, happy it wasn't the case.
This simulation is about an inside-out fire from autocombustion of a 10x10x4m garbage pile, but I'm unsure FDS is able to simulate a puctual smouldering fire located at the inner bottom level of the pile. I opted for assigning an HRRPUA curve exclusively to the upper surfaces as shown in the image, so the heat is indeed uniform as you mentioned in the last paragraph.

**Reply by Vojta-Salek** [reply]
Agreed, I would probably do the same (although I do not know all the details and aims of your work). Considering the size of your domain, it would be inappropriate to try to resolve in detail the combustion inside the large garbage pile. Therefore, you will be limited to the standard "layer 1D model", where material properties and thickness are assigned to an obstacle using a `&SURF` line. Even if the material was heated from backside, all combustibles would be automatically transported (more likely "teleported") to the material surface where combustion will take place.

**Reply by fr-evtech** [reply]
Yes, I can confirm I used the 1D model, and your take is completely correct. Also, I have to admit I'm currently not experienced enough to implement a 3D model, so it wasn't really a choice...
Thank you for your help and clarification. I have now two options, putting a solid phase device misuring the adiabatic surface temperature of the "burner" or a bunch of gas phase devices for radiative heat flux gas, after tweking a bit with the number of radiation angles.
I'll try both solutions.

**Comment by drjfloyd** [comment]
If you don't find it then you are not using the current release.

**Reply by fr-evtech** [reply]
You are right, sorry; found it.
